up_client_socket/rust/u_transport_socket.rs

```diff
@@ -0,0 +1,395 @@
+    ) -> Result<(), UStatus> {
+        dbg!("register listner called !");
+        if topic.authority.is_some() && topic.entity.is_none() && topic.resource.is_none() {
+            // This is special UUri which means we need to register for all of Publish, Request, and Response
+            // RPC response
+            Err(UStatus::fail_with_code(
+                UCode::UNIMPLEMENTED,
+                "Not implemented",
+            ))
+        } else {
+            // Do the validation
+            UriValidator::validate(&topic)
+                .map_err(|_| UStatus::fail_with_code(UCode::INVALID_ARGUMENT, "Invalid topic"))?;
+
+            if UriValidator::is_rpc_response(&topic) {
+                self.listner_map
+                    .lock()
+                    .unwrap()
+                    .entry(topic.to_string())
+                    .and_modify(|listener_local| listener_local.push(listener.clone()))
+                    .or_insert_with(|| vec![Arc::clone(&listener)as Arc<dyn UListener>]);
+
+                Ok(())
+            } else if UriValidator::is_rpc_method(&topic) {
+                self.listner_map
+                    .lock()
+                    .unwrap()
+                    .entry(topic.to_string())
+                    .and_modify(|listener_local| listener_local.push(listener.clone()))
+                    .or_insert_with(|| vec![Arc::clone(&listener) as Arc<dyn UListener>]);
+                dbg!("register listner called for rpc !");
+                Ok(())
+            } else {
+                self.listner_map
+                    .lock()
+                    .unwrap()
+                    .entry(topic.to_string())
+                    .and_modify(|listener_local| listener_local.push(listener.clone()))
+                    .or_insert_with(|| vec![Arc::clone(&listener) as Arc<dyn UListener>]);
+                dbg!("register listner called for topic !");
+                Err(UStatus::ok())
+            }
+        }
+    }
+
+    /// Unregisters a listener for a given topic.
+    ///
+    /// Messages arriving on this topic will no longer be processed by this listener.
+    ///
+    /// # Arguments
+    ///
+    /// * `topic` - Resolved topic uri where the listener was registered originally.
+    /// * `listener` - Identifier of the listener that should be unregistered.
+    ///
+    /// # Errors
+    ///
+    /// Returns an error if the listener could not be unregistered, for example if the given listener does not exist.
+    async fn unregister_listener(
+        &self,
+        topic: UUri,
+        listener: Arc<dyn UListener>,
+    ) -> Result<(), UStatus> {
+        let mut map = self.listner_map.lock().expect("Failed to acquire lock");
+        let listner_clone = Arc::clone(&listener) as Arc<dyn UListener>;
+
+        if let Some(listeners) = map.get_mut(&topic.to_string()) {
+            if let Some(index) = listeners
+                .iter()
+                .position(|l| Arc::ptr_eq(l, &listner_clone))
```

Comment:
If you take the suggestion [up here](https://github.com/eclipse-uprotocol/up-tck/pull/23/files#r1566424343):
You can:
```rust
let comparable_listener = ComparableListener::new(listener);
```
to then pull the relevant listener out via the hash without having to iterate.